src/Symfony/Component/Security/Http/Firewall.php

```diff
@@ -47,13 +47,17 @@ public function onKernelRequest(GetResponseEvent $event)
         }
 
         // register listeners for this firewall
-        list($listeners, $exceptionListener) = $this->map->getListeners($event->getRequest());
+        list($listeners, $exceptionListener, $logoutListener) = $this->map->getListeners($event->getRequest());
```

Comment:
This is probably what the build fails on. The return type is basically a [Tuple](https://en.wikipedia.org/wiki/Tuple).

I'm assuming that `list()`doesn't handle smaller arrays well. You can fix this with:

```
$firewallListeners = $this->map->getListeners($event->getRequest());
$listeners = $firewallListeners[0];
$exceptionListener = $firewallListeners[1];

if (\count($listeners) !== 3) {
    @trigger_error('As of Symfony 4.2 it is deprecated to pass only the listeners and exception listener from the firewall context. Provide either a logout listener or null.', E_USER_DEPRECATED);
}
$logoutListener = $firewallListeners[1] ?? null;
```

Message might need tweaking.

Reply:
But I updated mocks by taking this into account (like https://github.com/symfony/symfony/pull/24805/files#diff-1da8f667e9085d5ed87adf1c41be9ed6L211) so all tuples should have three elements?

Reply:
Yes I think the issue comes from a mock but I fail to see which one.